proxy/src/main/java/com/velocitypowered/proxy/protocol/packet/Respawn.java

```diff
@@ -74,33 +114,57 @@ public String toString() {
         + ", difficulty=" + difficulty
         + ", gamemode=" + gamemode
         + ", levelType='" + levelType + '\''
+        + ", shouldKeepPlayerData=" + shouldKeepPlayerData
+        + ", isDebug=" + isDebug
+        + ", isFlat='" + isFlat
+        + ", dimensionRegistryName='" + dimensionRegistryName + '\''
         + '}';
   }
 
   @Override
   public void decode(ByteBuf buf, ProtocolUtils.Direction direction, ProtocolVersion version) {
-    this.dimension = buf.readInt();
+    if (version.compareTo(ProtocolVersion.MINECRAFT_1_16) >= 0) {
+      this.dimensionRegistryName = ProtocolUtils.readString(buf); // Not sure what the cap on that is
+    } else {
+      this.dimension = buf.readInt();
+    }
     if (version.compareTo(ProtocolVersion.MINECRAFT_1_13_2) <= 0) {
       this.difficulty = buf.readUnsignedByte();
     }
     if (version.compareTo(ProtocolVersion.MINECRAFT_1_15) >= 0) {
       this.partialHashedSeed = buf.readLong();
     }
     this.gamemode = buf.readUnsignedByte();
-    this.levelType = ProtocolUtils.readString(buf, 16);
+    if (version.compareTo(ProtocolVersion.MINECRAFT_1_16) >= 0) {
+      isDebug = buf.readBoolean();
+      isFlat = buf.readBoolean();
+      shouldKeepPlayerData = buf.readBoolean();
```

Comment:
Minor nit: inconsistent use of `this`. Either use it for setting all fields, or use it for none.